wandb/sdk/lib/service_connection.py

```diff
@@ -0,0 +1,163 @@
+import atexit
+import os
+from typing import Optional
+
+from wandb.proto import wandb_internal_pb2 as pb
+from wandb.proto import wandb_server_pb2 as spb
+from wandb.proto import wandb_settings_pb2
+from wandb.sdk.interface.interface import InterfaceBase
+from wandb.sdk.interface.interface_sock import InterfaceSock
+from wandb.sdk.lib import service_token
+from wandb.sdk.lib.exit_hooks import ExitHooks
+from wandb.sdk.lib.mailbox import Mailbox
+from wandb.sdk.lib.sock_client import SockClient
+from wandb.sdk.service import service
+
+
+class WandbServiceNotOwnedError(Exception):
+    """Raised when the current process does not own the service process."""
+
+
+def connect_to_service(
+    settings: "wandb_settings_pb2.Settings",
```

Comment:
why do you need quotes here?

Reply:
https://peps.python.org/pep-0563/

> [PEP 3107](https://peps.python.org/pep-3107/) added support for arbitrary annotations on parts of a function definition. Just like default values, annotations are evaluated at function definition time. This creates a number of issues for the type hinting use case:
>
> [...]
>
> * type hints are executed at module import time, which is not computationally free.

I removed all quotes now and added a `from future import __annotations__` instead. This has the added benefit of allowing the modern `|` syntax instead of `Union` and `Optional`.